plugin/pkg/auth/authenticator/token/tokenfile/tokenfile.go

```diff
@@ -63,10 +81,58 @@ func NewCSV(path string) (*TokenAuthenticator, error) {
 			obj.Groups = strings.Split(record[3], ",")
 		}
 	}
+	return tokens, nil
+}
 
-	return &TokenAuthenticator{
-		tokens: tokens,
-	}, nil
+func (a *TokenAuthenticator) Update(tokens map[string]*user.DefaultInfo) {
+	a.tokens = tokens
+}
+
+func (a *TokenAuthenticator) AddFileWatcher(ready *chan bool) error {
```

Comment:
Please do not copypaste this function all over, put it in a common place (find or create an appropriate subpackage under pkg/util/) and parameterize appropriately.